### ISSUE TITLE
Update to Python 3.10

### ISSUE THREAD

**Issue by jwhitlock** [mozilla/ichnaea#1746]
[Python 3.10](https://docs.python.org/3/whatsnew/3.10.html) has some nice changes in syntax errors, some deprecation warnings, and some optimizations. I think it should be safe to update ichnaea to use it.

Currently, it fails because shapely doesn't have speedups available with Python 3.10 (see https://github.com/shapely/shapely/issues/1263).

**Comment by jwhitlock** [comment]
I disabled the speedups check, and otherwise Python 3.10.2 works.

Looking at the code, the Geocoder `GEOCODER` is primarily used to turn a lat/long estimate (location) into a region, which could be the places it is used in the API. The last time I investigated, this functionality was not used by Firefox, and insignificantly by other clients. However, I can't say these code paths are unused, and they appear to be used in the backend to assign a country to a station.

So, to play it safe, I'm going to continue to wait for Shapely 1.8.1 to update to Python 3.10.
